ballerina/commons.bal

```diff
@@ -16,6 +16,47 @@
 
 import ballerina/time;
 
+# Represents the task service.
+public type Service distinct service object {
+  isolated function onTrigger() returns error?;
```

Comment:
Any reason to use the generic `error` here ? I think `onTrigger` should return a module specific error rather than a generic error.

Reply:
This change will affect the implementation of this function right? User can return any type of error from the implementation of this function

Reply:
We are currently logging that error

Reply:
We could add an onError method. But I think logging would be ok for the initial release. Lets create an issue for the onError method

Reply:
`onError` is there to be invoked when there is an error while invoking the `onTrigger` method (in this case). So, it is not required unless we have specific errors which can be thrown while invoking the `onTrigger` method. 

> We are currently logging that error

+1